packages/react-jss/src/JssProvider.js

```diff
@@ -20,85 +20,88 @@ type Props = {
   generateId?: GenerateId,
   classNamePrefix?: string,
   disableStylesGeneration?: boolean,
+  sheetOptions: StyleSheetFactoryOptions,
   children: Node
 }
 
 export default class JssProvider extends Component<Props> {
   static propTypes = {
-    ...propTypes,
+    registry: PropTypes.instanceOf(SheetsRegistry),
+    jss: PropTypes.shape({}),
     generateId: PropTypes.func,
     classNamePrefix: PropTypes.string,
     disableStylesGeneration: PropTypes.bool,
-    children: PropTypes.node.isRequired
+    children: PropTypes.node.isRequired,
+    sheetOptions: PropTypes.shape({})
   }
 
-  static childContextTypes = contextTypes
+  managers: {} = {}
 
-  static contextTypes = contextTypes
+  generateId = createGenerateId()
```

Comment:
Now you will create a new generator every time user renders a new provider, right? Is it what you wanted to achieve?

Reply:
I will create a generator for every rendered instance.

Reply:
yes but that will mean that 2 instances will generate conflicting ids